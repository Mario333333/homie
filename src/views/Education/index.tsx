import { useContext } from "react";
import book from "../../assets/svg/book.svg";
import empty_image from "../../assets/images/empty_image.png";
import ellipse from "../../assets/svg/ellipse.svg";
import InformationElement from "../../components/InformationElement";
import LicensesCertifications from "../../components/LicensesCertifications";
import useAxios from "../../hooks/useAxios";
import app_context from "../../app_context";
import {
  InformationElementData,
  LicensesCertificationsData,
} from "../../utils/ts/componentTypes";
import Context from "../../utils/ts/contextType";

const Education = () => {
  const context: Context = useContext(app_context);
  const { labels } = context;
  const { data: informationElementData, error: educationError } =
    useAxios<Array<InformationElementData>>("education");
  const { data: certificationsData, error: certificationsError } =
    useAxios<Array<LicensesCertificationsData>>("certifications");

  return (
    <div className="education-container">
      {educationError === false &&
        informationElementData !== null &&
        informationElementData.map(
          (item: InformationElementData, index: number) => {
            return (
              <InformationElement
                icon={index > 0 ? ellipse : book}
                item={item}
                type="school"
                key={`school-${item.id}`}
              />
            );
          }
        )}
      <h2 className="title">{labels?.licenses_certifications}</h2>
      {certificationsError === false &&
        certificationsData !== null &&
        certificationsData.map(
          (item: LicensesCertificationsData, index: number) => {
            return (
              <LicensesCertifications
                key={`LC-${index}`}
                item={item}
                image={empty_image}
              />
            );
          }
        )}
    </div>
  );
};

export default Education;
