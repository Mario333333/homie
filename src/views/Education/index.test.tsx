import Education from "./index";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup, waitFor } from "@testing-library/react";
import { Router, Route, Switch } from "react-router-dom";
import { createMemoryHistory } from "history";
import Context from "../../app_context";

const history = createMemoryHistory();

describe("Education", () => {
  afterEach(cleanup);
  const labels = {
    about_me: "Acerca de mi",
    contact: "Contacto",
    education: "EducaciÃ³n",
    experience: "Experiencia",
    portfolio: "Portafolio",
    knowledge_skills: "Conocimientos y habilidades",
    languages: "Idiomas",
    skills: "Habilidades",
    licenses_certifications: "Licencias y certificaciones",
    interests: "Intereses",
  };

  test("education snapshot", () => {
    const { asFragment } = render(<Education />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("education defined component and content", async () => {
    history.push("/");
    const component = render(
      <Context.Provider value={{ labels }}>
        <Router history={history}>
          <Switch>
            <Route exact path="/">
              <Education />
            </Route>
          </Switch>
        </Router>
      </Context.Provider>
    );

    await waitFor(() => {
      expect(component).toBeDefined();
      expect(component.getByText(labels.licenses_certifications)).toBeDefined();
    });
  });
});
