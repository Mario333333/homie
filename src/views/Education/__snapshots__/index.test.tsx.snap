// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education education snapshot 1`] = `
<DocumentFragment>
  <div
    class="education-container"
  >
    <h2
      class="title"
    />
  </div>
</DocumentFragment>
`;
